lib/data/dataset.js

```diff
@@ -82,6 +82,14 @@ const getDataset = (xml) =>
     else if (!semverSatisfies(version.get(), '2022.1.0 - 2024.1.x'))
       throw Problem.user.invalidEntityForm({ reason: `Entities specification version [${version.get()}] is not supported.` });
 
+    let warnings;
+    if (semverSatisfies(version.get(), '<2024.1.x'))
+      warnings = {
```

Comment:
The name `warnings` in the plural makes me think that it should be an array. Just a thought, I also wouldn't mind leaving it as-is until we have a second warning.